src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetRandomCommand.cs

```diff
@@ -532,7 +545,7 @@ protected override void EndProcessing()
                     WriteObject(_chosenListItems[j]);
 
                     // remove the output object from consideration in the next iteration.
-                    if (i != j)
+                    if (i != j && ! _unique)
```

Comment:
And then you'd just use `Unique` here rather than the backing property. 🙂 

Also, going by the name of the property I'd expect this condition to be reversed? If I call `1..10 | Get-Random -Unique` I'd expect only unique entries, not the disabling of the unique selection behaviour.

Either the parameter name or the behaviour should be switched, though the former is easier with respect to maintaining compatibility.

Reply:
That was what I intended to do, something got twisted in my brain, :-)